lib/src/neu/types.ts

```diff
@@ -302,6 +338,10 @@ export function isIntLiteralType(type: Type): type is IntLiteralType {
   return type.kind === TypeKind.INT_LITERAL;
 }
 
+export function areIntLiteralTypes(types: Type[]): types is IntLiteralType[] {
+  return types.every(isIntLiteralType);
```

Comment:
Same suggestion here - the logic is duplicated. You could encapsulate this logic and pass the predicate (i.e. `isIntLiteralType`) as a parameter. 
`areLiteralTypes = (types, predicate) => types.every(predicate)`